cwms-data-api/src/main/java/cwms/cda/data/dao/watersupply/WaterContractDao.java

```diff
@@ -0,0 +1,293 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.data.dao.watersupply;
+
+import static java.util.stream.Collectors.toList;
+
+import cwms.cda.data.dao.JooqDao;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.LookupType;
+import cwms.cda.data.dto.watersupply.PumpType;
+import cwms.cda.data.dto.watersupply.WaterUser;
+import cwms.cda.data.dto.watersupply.WaterUserContract;
+import java.util.ArrayList;
+import java.util.List;
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+import usace.cwms.db.dao.ifc.loc.LocationRefType;
+import usace.cwms.db.dao.ifc.watersupply.WaterUserContractRefType;
+import usace.cwms.db.dao.ifc.watersupply.WaterUserContractType;
+import usace.cwms.db.dao.ifc.watersupply.WaterUserType;
+import usace.cwms.db.dao.util.OracleTypeMap;
+import usace.cwms.db.jooq.codegen.packages.CWMS_WATER_SUPPLY_PACKAGE;
+import usace.cwms.db.jooq.codegen.udt.records.LOCATION_REF_T;
+import usace.cwms.db.jooq.codegen.udt.records.LOOKUP_TYPE_TAB_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_CONTRACT_REF_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_CONTRACT_TAB_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_OBJ_T;
+import usace.cwms.db.jooq.codegen.udt.records.WATER_USER_TAB_T;
+import usace.cwms.db.jooq.dao.util.LocationTypeUtil;
+import usace.cwms.db.jooq.dao.util.LookupTypeUtil;
+import usace.cwms.db.jooq.dao.util.WaterUserTypeUtil;
+
+
+
+public class WaterContractDao extends JooqDao<WaterUserContract> {
+    public WaterContractDao(DSLContext dsl) {
+        super(dsl);
+    }
+
+    public List<WaterUserContract> getAllWaterContracts(CwmsId projectLocation, String entityName) {
+        List<WaterUserContract> retVal = new ArrayList<>();
+        List<WaterUserContractType> waterUserContractTypes = new ArrayList<>();
+
+        LocationRefType locationRefType = WaterSupplyUtils.map(projectLocation);
+
+        waterUserContractTypes.addAll(connectionResult(dsl, c -> {
+            setOffice(c, projectLocation.getOfficeId());
+            LOCATION_REF_T projectLocationRef =  LocationTypeUtil.toLocationRefT(locationRefType);
+            WATER_USER_CONTRACT_TAB_T waterUserContractObjTs = CWMS_WATER_SUPPLY_PACKAGE.call_RETRIEVE_CONTRACTS(
+                    DSL.using(c).configuration(), projectLocationRef, entityName);
+            return waterUserContractObjTs.stream()
+                    .map(WaterUserTypeUtil::toWaterUserContractType)
+                    .collect(toList());
+        }));
+
+        for (WaterUserContractType waterUserContractType : waterUserContractTypes) {
```

Comment:
The mapping can be handled within the connectionResult block

Reply:
Altered mapping as suggested